plugins/onepad/Linux/dialog.cpp

```diff
@@ -391,11 +391,14 @@ void on_view_key_clicked(GtkToggleButton *togglebutton, gpointer user_data)
 void on_toggle_option(GtkToggleButton *togglebutton, gpointer user_data)
 {
 	dialog_checkbox *checkbox = (dialog_checkbox*)user_data;
+	u32 joyid = conf->get_joyid(current_pad);
 
 	if (gtk_toggle_button_get_active(togglebutton))
 		conf->options |= checkbox->mask;
 	else
 		conf->options &= ~checkbox->mask;
+
+	s_vjoysticks[joyid]->Init(joyid);
```

Comment:
Because the number of buttons the joystick has is calculated at Init() time. This will re-init the joystick object when the hack is turned off or on.  This is a convenience as you would have to exit the dialog and re-enter every time you toggle the hack otherwise.  Since I stuck this in the on_toggle_option() function, it will actually re-init the object when any checkbox is toggled.  You could put a check on the Init() call to only call when its the hack checkbox, but as is it doesn't hurt anything.